.github/actions/setup-build/action.yml

```diff
@@ -0,0 +1,120 @@
+name: "Cached build"
+description: "Setup the build using cache"
+inputs:
+  ghc:
+    description: "Ghc version"
+    required: true
+  cabal:
+    description: "Cabal version"
+    required: false
+    default: "3.6"
+  os:
+    description: "Operating system: Linux, Windows or macOS"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - uses: haskell/actions/setup@v1
+      id: HaskEnvSetup
+      with:
+        ghc-version  : ${{ inputs.ghc   }}
+        cabal-version: ${{ inputs.cabal }}
+        enable-stack: false
+
+    - if: inputs.os == 'Windows'
+      name: (Windows) Platform config
+      run: |
+        echo "CABAL_PKGS_DIR=C:\\cabal\\packages" >> $GITHUB_ENV
+      shell: bash
+    - if: ( inputs.os == 'Linux' ) || ( inputs.os == 'macOS' )
+      name: (Linux,macOS) Platform config
+      run: |
+        echo "CABAL_PKGS_DIR=~/.cabal/packages" >> $GITHUB_ENV
+      shell: bash
+
+    # Needs to be before Cache Cabal so the cache can detect changes to the modified cabal.project file
+    - if: inputs.ghc == '9.0.1' || inputs.ghc == '9.2.1'
```

Comment:
We could consider making this unconditional and just looking for the existence of a suffixed cabal.project for the current version. Then this would work automatically if we add/remove them in future. Perhaps too magical?

Reply:
Yeah, i considered it at somepoint but i forgot that possibility, thanks!